WordPress/Classes/Extensions/String+RegEx.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+
+
+// MARK: - String: RegularExpression Helpers
+//
+extension String {
+
+    func matches(regex: String, options: NSRegularExpression.Options = []) -> [NSTextCheckingResult] {
+        let regex = try! NSRegularExpression(pattern: regex, options: options)
```

Comment:
So... tbh I don't think these methods should try to do anything with the exception.  In fact I think I should change these methods to just throw whatever the base method throws, so that the code outside of these can decide what to do with the exceptions.